### ISSUE TITLE
Angus and Jakarta mail dependency conflict

### ISSUE THREAD

**Issue by Euquimides** [craftercms/craftercms#7391]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] The issue is in the develop branch
- [ ] The issue is in the latest released 4.1.x

### Describe the issue

As shown here: https://spring.io/blog/2023/07/19/spring-integrtion-6-2-milestone-1-available

Spring 6 added Angus as dependency, changing `org.eclipse.angus:jakarta.mail` as Java Mail API implementation which replaces `com.sun.mail.imap` package with the `org.eclipse.angus.mail.imap`. Since we also have `jakarta` as dependency, there might be a conflict with both packages, affecting Crafter's email service as such. 

We should check which dependency to quit so to avoid any runtime errors with the email service. 

### Steps to reproduce

Steps:

1. Initialize Crafter Studio. 
2. In the login screen go to `Forgot your password?`
3. Add the username (in this case the admin username)
4. Send the request and see the issue: no email is being sent. 

### Relevant log output

```shell
DEBUG: Jakarta Mail version 2.1.3
DEBUG: URL jar:file:/home/juanvasquez/crafter/craftercms-copy/craftercms/crafter-authoring/bin/apache-tomcat/webapps/studio/WEB-INF/lib/angus-mail-1.0.0.jar!/META-INF/javamail.providers
Exception in thread "forgotPasswordTaskExecutor-3" java.lang.IncompatibleClassChangeError: Class com.sun.mail.util.LineInputStream does not implement the requested interface jakarta.mail.util.LineInputStream
	at jakarta.mail.Session.loadProvidersFromStream(Session.java:1065)
	at jakarta.mail.Session.access$000(Session.java:204)
	at jakarta.mail.Session$2.load(Session.java:965)
	at jakarta.mail.Session.loadAllResources(Session.java:1252)
	at jakarta.mail.Session.loadProviders(Session.java:1002)
	at jakarta.mail.Session.<init>(Session.java:274)
	at jakarta.mail.Session.getInstance(Session.java:327)
	at org.springframework.mail.javamail.JavaMailSenderImpl.getSession(JavaMailSenderImpl.java:162)
	at org.springframework.mail.javamail.JavaMailSenderImpl.createMimeMessage(JavaMailSenderImpl.java:335)
	at org.craftercms.studio.impl.v2.security.password.EmailForgotPasswordTaskFactory.sendEmail(EmailForgotPasswordTaskFactory.java:135)
	at org.craftercms.studio.impl.v2.security.password.EmailForgotPasswordTaskFactory.forgotPassword(EmailForgotPasswordTaskFactory.java:123)
	at org.craftercms.studio.impl.v2.security.password.EmailForgotPasswordTaskFactory.lambda$prepareTask$0(EmailForgotPasswordTaskFactory.java:94)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
```

**Comment by sumerjabri** [comment]
Look into `greenmail` in `Crafter Social` as well.

**Comment by Euquimides** [comment]
I just checked email reset-password with Crafter Profile (which I think uses also `Greenmail`), seems it is working. Still worth checking out `Greenmail` implementation in case of any changes. 

![Screenshot from 2024-10-14 12-40-59](https://github.com/user-attachments/assets/9faf4c7e-4d34-49ca-8a93-2716bc642158)

**Comment by Euquimides** [comment]
Hi! Updated and rechecked this issue. Tested this in Social, Profile, Authoring, Delivery and Deployer. Fix verified for v4.2.0. Thank you! Closing this ticket.
